pkg/snet/network.go

```diff
@@ -81,16 +82,28 @@ type Network struct {
 }
 
 // New creates a network from a config.
-func New(conf Config) *Network {
+func New(conf Config, eb *appevent.Broadcaster) *Network {
 	var dmsgC *dmsg.Client
 	var stcpC *stcp.Client
 
 	if conf.Dmsg != nil {
-		c := &dmsg.Config{
+		dmsgConf := &dmsg.Config{
 			MinSessions: conf.Dmsg.SessionsCount,
+			Callbacks: &dmsg.ClientCallbacks{
+				OnSessionDial: func(network, addr string) error {
+					data := appevent.TCPDialData{RemoteNet: network, RemoteAddr: addr}
+					event := appevent.NewEvent(appevent.TCPDial, data)
+					_ = eb.Broadcast(context.Background(), event) //nolint:errcheck
```

Comment:
Why don't we return an error here? Wondering, cause the func signature allows that

Reply:
@Darkren If an error is returns, the session dial is canceled. This may not be intended behavior. I'll write a comment.